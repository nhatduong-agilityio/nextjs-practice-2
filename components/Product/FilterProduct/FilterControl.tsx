import { Button, Heading, Stack } from '@chakra-ui/react'
import React, { memo } from 'react'

interface IProps {
  onHandleOpenFilter: () => void
}

const FilterControl = memo(({ onHandleOpenFilter }: IProps) => (
  <Stack flexDir='row' justifyContent='space-between' alignItems='center' mb={['20px', '50px']}>
    <Heading variant={['extraBoldMD', 'extraBoldXL']}>Product</Heading>
    <Button
      display={['flex', 'none']}
      color='black'
      variant='ghost'
      fontSize='14px'
      width='max-content'
      height='fit-content'
      style={{
        paddingInline: 0,
      }}
      onClick={onHandleOpenFilter}
      justifyContent='flex-end'
    >
      Filter
    </Button>
  </Stack>
))

FilterControl.displayName = 'FilterControl'

export default FilterControl
