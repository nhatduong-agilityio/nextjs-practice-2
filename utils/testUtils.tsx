import { ChakraProvider } from '@chakra-ui/react'
import { AuthProvider } from '@store/providers/AuthProvider'
import { LoadingProvider } from '@store/providers/LoadingProvider'
import type { RenderOptions, RenderResult } from '@testing-library/react'
import { render } from '@testing-library/react'
import { customTheme } from '@themes/index'
import type { ReactElement } from 'react'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <ChakraProvider theme={customTheme}>
      <LoadingProvider>
        <AuthProvider>{children}</AuthProvider>
      </LoadingProvider>
    </ChakraProvider>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
