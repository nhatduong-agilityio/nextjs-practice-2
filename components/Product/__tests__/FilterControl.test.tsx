// Utils
import { render } from '@utils/testUtils'

// Components
import FilterControl from '../FilterProduct/FilterControl'

jest.mock('next/navigation', () => ({
  useSearchParams: jest.fn(),
}))

describe('FilterControl render', () => {
  const propsMock = {
    onHandleOpenFilter: jest.fn(),
  }

  it('Should FilterControl match Snapshot', () => {
    const { container } = render(<FilterControl {...propsMock} />)

    expect(container).toMatchSnapshot()
  })
})
