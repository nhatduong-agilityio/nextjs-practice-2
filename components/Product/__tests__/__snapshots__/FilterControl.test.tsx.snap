// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterControl render Should FilterControl match Snapshot 1`] = `
<div>
  <div
    class="chakra-stack css-blzuji"
  >
    <h2
      class="chakra-heading css-4t11uo"
    >
      Product
    </h2>
    <button
      class="chakra-button css-10zc6up"
      style="padding-inline: 0;"
      type="button"
    >
      Filter
    </button>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
